src/detect_breakends.cpp

```diff
@@ -6,34 +6,60 @@ struct cmd_arguments
 {
     std::filesystem::path alignment_file_path{};
     std::filesystem::path insertion_file_path{};
+    std::vector<uint8_t> methods{1, 2, 3, 4};   // default is using all methods
 };
 
 void initialize_argument_parser(seqan3::argument_parser & parser, cmd_arguments & args)
 {
-    parser.info.author = "David Heller";
+    parser.info.author = "David Heller & Lydia Buntrock";
+    parser.info.app_name = "iGenVar";
+    parser.info.man_page_title = "Short and long Read SV Caller";
     parser.info.short_description = "Detect junctions in a read alignment file";
     parser.info.version = "0.0.1";
+    parser.info.date = "19-01-2021";    // last update
+    parser.info.email = "lydia.buntrock@fu-berlin.de";
+    parser.info.long_copyright = "long_copyright";
+    parser.info.short_copyright = "short_copyright";
+    parser.info.url = "https://github.com/seqan/iGenVar/";
+
+    // Validatiors:
+    // seqan3::value_list_validator method_validator{"1", "cigar_string",
+    //                                               "2", "split_read",
+    //                                               "3", "read_pairs",
+    //                                               "4", "read_depth"};
+    seqan3::arithmetic_range_validator method_validator{1, 4};
+
+    // Options - Input / Output:
     parser.add_positional_option(args.alignment_file_path, "Input read alignments in SAM or BAM format.",
                                  seqan3::input_file_validator{{"sam", "bam"}} );
     parser.add_positional_option(args.insertion_file_path, "Output file for insertion alleles",
                                  seqan3::output_file_validator{{"fa", "fasta"}} );
+
+    // Options - Methods:
+    parser.add_option(args.methods, 'm', "method", "Choose the method to be used.",
+                      seqan3::option_spec::ADVANCED, method_validator);
```

Comment:
The way this is, a user could only choose one of the four methods, no? E.g. you can't use `-m 1, 2, 3, 4` or something. Maybe instead, it's better to have four flags instead of options? Then the user could do something like `-c -s -d` for cigar, split reads, and read depth.

Although to be completely honest, I can't really think of a reason why a user would not want to use the CIGAR string. For split reads, if the user has very short reads I guess they might think split reads aren't useful. Same for read depth if the sequencing was very shallow.

With regards to read pairs, I think this is something we should detect automatically (whether the sequencing was done with single end or paired end sequencing), so that a user doesn't accidentally turn on the read pairs option with single end sequencing data.

Reply:
I think with the current implementation, a user can choose four methods using something like `-m 1 -m 2 -m 3 -m 4` because args.methods is a `std::vector`.

You have a good point, though, with the CIGAR strings and the read pairs. It might be good to spend some time thinking about efficient processing of the SAM/BAM files:

- cigar and split reads collect evidence from each individual read.
- read depth, in contrast, is an analysis performed on the depth profile along the genomes, not individual reads.
- read pair should (like you say) only be used for paired-end data. But for this type of data it works on a per-read basis like cigar and split reads.

So we should maybe have a separate detection function for read depth because it's completely separate from the other methods.

Reply:
I think my change (the modularisation) is there to compare all methods with all methods. 
But later, such a recognition from the SAM/BAM files would certainly be useful. I would create a new issue for this.
And then you could make some methods dependent on the input files.